File: 백준/Bronze/27433. 팩토리얼 2/팩토리얼 2.js
const fs = require("fs");
const filePath = process.platform === "linux" ? "/dev/stdin" : "./input.txt";
const input = fs.readFileSync(filePath).toString().trim().split(" ");

const dp = [];
Solution = (N) => {
    dp[0] = 1;
    dp[1] = 1;
    for (i = 2; i <= N; i++) {
        dp[i] = dp[i - 1] * i;
    }
    console.log(dp[N]);
};

Solution(input[0]);

File: 백준/Bronze/27433. 팩토리얼 2/팩토리얼 2.cc
#include <iostream>
#include <string>
#include <cstring>
#include <vector>
#include <tuple>
#include <algorithm>
#include <cmath>
#define endl "\n"
using namespace std;

long long dp[21];

void Solution(int N)
{
	dp[0] = 1;
	dp[1] = 1;

	for (int i = 2; i <= N; i++)
	{
		dp[i] = dp[i - 1] * i;
	}

	cout << dp[N];
}
int main()
{
	//freopen("input.txt", "r", stdin);
	ios::sync_with_stdio(false);
	cin.tie(NULL);
	cout.tie(NULL);

	// input 입력
	int N;
	cin >> N;

	Solution(N);
	return 0;
}